code/modules/antagonists/changeling/abilities/_changeling_ability_holder.dm

```diff
@@ -172,6 +172,12 @@
 				return 1//what could possibly go wrong
 		return 0
 
+	cast(atom/target)
+		. = ..()
+		if (!isturf(src.holder.owner.loc) && !src.can_use_in_container)
+			boutput(src.holder.owner, "<span class='alert'>You can't use this ability here.</span>")
+			return TRUE
+
```

Comment:
this should be in `castcheck` I think

Reply:
Works here, which is why I left it, does it matter?

Reply:
Right but the checks should be in the check proc, otherwise you'll get issues with code assuming the ability has successfully cast when it hasn't.